kbtree.h

```diff
@@ -359,7 +359,37 @@ typedef struct {
 			if (itr->p < itr->stack) return 0; \
 			if (itr->p->x && itr->p->i < itr->p->x->n) return 1; \
 		} \
-	}
+	} \
+	static inline int kb_itr_prev_##name(kbtree_##name##_t *b, kbitr_t *itr) \
+	{ \
+		if (itr->p < itr->stack) return 0; \
+		for (;;) { \
+			while (itr->p->x && itr->p->i >= 0) { \
+				itr->p[1].x = itr->p->x->is_internal? __KB_PTR(b, itr->p->x)[itr->p->i] : 0; \
+				itr->p[1].i = itr->p[1].x ? itr->p[1].x->n : -1; \
+				++itr->p; \
+			} \
+			--itr->p; \
+			if (itr->p < itr->stack) return 0; \
+			--itr->p->i; \
+			if (itr->p->x && itr->p->i >= 0) return 1; \
+		} \
+	} \
+	static int kb_itr_getp_##name(kbtree_##name##_t *b, const key_t * __restrict k, kbitr_t *itr) \
```

Comment:
How about adding inline here? The compiler will trigger `defined but not used` warnings when there is no inline.
The above definition has the same problem:
```
static int kb_itr_get_##name(
```